src/vendor_upstream/core/shallowEqual.js

```diff
@@ -0,0 +1,49 @@
+/**
```

Comment:
I deleted this on v0.8 since it became unused, but we merged in a master change that uses it again, so it's back